src/pymor/std_lib/time_bounds.py

```diff
@@ -0,0 +1,202 @@
+
+    Returns
+    -------
+    xr.Dataset
+        The output dataset with the time bounds information.
+    """
+    # Get dataset name for logging
+    dataset_name = ds.attrs.get("name", "unnamed_dataset")
+
+    # Log header with markdown style
+    logger.info(f"[Time bounds] {dataset_name}")
+    logger.info(f"  → is dataset:   {'✅' if isinstance(ds, xr.Dataset) else '❌'}")
+
+    # Check if input is a dataset
+    if not isinstance(ds, xr.Dataset):
+        logger.error("  ❌ The input is not a dataset.")
+        raise ValueError("The input is not a dataset.")
+
+    # Get time label and check if it exists
+    time_label = get_time_label(ds)
+    if time_label is None:
+        logger.error("  ❌ The dataset does not contain a valid time coordinate.")
+        raise ValueError("The dataset does not contain a valid time coordinate.")
+
+    bounds_dim_label = "bnds"
+    time_bounds_label = f"{time_label}_{bounds_dim_label}"
+
+    # Log time and bounds info
+    logger.info(f"  → time label  : {time_label}")
+    logger.info(f"  → bounds label: {bounds_dim_label}")
+
+    # Get approx_interval from rule (in days)
+    approx_interval = getattr(rule, "approx_interval", None)
+    logger.info(f"  → approx_interval: {approx_interval} days")
+
+    # Get time method from rule or dataset attributes
+    time_method = getattr(rule, "time_method", None) or ds.attrs.get(
+        "time_method", "mean"
+    )
+    logger.info(f"  → time method : {time_method}")
+
+    # Check if time bounds already exist
+    has_time_bounds = time_bounds_label in ds.variables
+    logger.info(f"  → has time bounds: {'✅' if has_time_bounds else '❌'}")
+
+    if has_time_bounds:
+        logger.info(f"  → using existing bounds: {time_bounds_label}")
+        return ds
+
+    # Validate time method
+    if time_method not in ["mean", "instantaneous", "climatology"]:
+        logger.warning(
+            f"  ⚠️  Unknown time method '{time_method}', defaulting to 'mean'"
+        )
+        time_method = "mean"
+
+    # Only create bounds for mean and instantaneous methods
+    if time_method == "climatology":
+        logger.info("  → skipping bounds creation for climatology data")
+        return ds
+
+    # Get time coordinate
+    time_var = ds[time_label]
+    time_values = time_var.values
+
+    # Check if we have enough time points
+    if len(time_values) < 1:
+        error_msg = "Cannot create time bounds: no time points found"
+        logger.error(f"  ❌ {error_msg}")
+        raise ValueError(error_msg)
+
+    # Log time values info
+    logger.info(
+        f"  → time values  : {len(time_values)} points from {time_values[0]} to {time_values[-1]}"
+    )
+
+    # Handle instantaneous time method
+    if time_method == "instantaneous":
+        logger.info("  → creating instantaneous bounds (delta time = 0)")
+        # For instantaneous data, bounds are the same as time values
+        bounds_data = np.column_stack([time_values, time_values])
```

Comment:
As a reminder of what `np.column_stack` does:
```suggestion
        # For instantaneous data, bounds are the same as time values
        #
        # .. note:: np.column_stack reminder
        #
        #    >>> a = np.array([1, 2, 3])
        #    >>> b = np.array([4, 5, 6])
        #    >>> np.column_stack((a, b))
        #    [[1, 4], [2, 5], [6, 3]]
        #
        bounds_data = np.column_stack([time_values, time_values])
```